### PR TITLE
Refactor, bugfixes, additional tests.

### PR DESCRIPTION
**Pull request by peterldowns**
Now at 96.5% coverage for the entire package; 100% coverage for `vangoh.go` itself. The only un-tested code is `vangoh.NegroniHandler`.

Changes include:
* Easier-to-type variable naming.
* Consistent use of Key and Secret across the codebase.
* Adding more tests, greatly increasing coverage.
* Exporting helper functions to make it easier for consumers of the library to test their integration.
* Not reporting error information by default, but allowing exposure via a debug flag.
* Compiling route regexes at time of inclusion, rather than re-compiling per-request.
* Allowing secret providers to store custom data and access the request immediately after successful authentication, useful when writing custom auth middlewares.
* No longer decodes the signature sent by the client, simply creates a b64 signature on the server and the compares them.

Would be happy to talk about these changes individually, break them into separate PRs as necessary, etc.

### CODE REVIEW

**Comment by peterldowns** [comment]
@twrobel3 ETA on a review? Days; weeks; months? :stuck_out_tongue_winking_eye:

File: dependencies.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+)
+
+type Clock struct {
+	Now func() time.Time
+}
+
+var clock = Clock{}
+
+// Set up default implementations of all dependencies.
+// Tests can override these implementations as necessary.
```
Comment (twrobel3) [inline dependencies.go]: This seems to only serve a purpose in testing, as the only time it's run outside of test code is when calling `present := clock.Now()`.  If this is the case, can we make the Clock struct un-exported?

Also, the name "dependencies" for this file is confusing

Comment (peterldowns) [inline dependencies.go]: Good point – moved to an unexported variable in `vangoh.go`.

**Comment by peterldowns** [comment]
Any other feedback / changes required before merging?

**Comment by twrobel3** [comment]
Nope, looks good to go.  Merged!

Cheers,
-Taylor

**Comment by peterldowns** [comment]
Thanks!

**Pull request merged**